java/src/shared/model/game/Game.java

```diff
@@ -826,13 +835,22 @@ public void rollNumber(final int value) throws Exception {
         //Is value a 7 - robber
         if (value == 7) {
             //Go to discarding phase before robbing if any player has to discard
-            getPlayers().forEach(player -> {
+            boolean discard = false;
+            for(Player player: getPlayers()){
                 if (player.canDiscardCards()) {
```

Comment:
besides, the client makes it impossible to discard twice anyway.  so you are good corbin